autoload/lsp/ui/vim/semantic.vim

```diff
@@ -62,28 +63,61 @@ function! s:init_highlight(server, buf) abort
 
             silent! call prop_type_add(s:get_textprop_name(a:server, l:scope_idx), {'bufnr': a:buf, 'highlight': l:hl, 'combine': v:true})
         endfor
+
+        silent! call prop_type_add(s:textprop_cache, {'bufnr': a:buf})
     endif
 
     call setbufvar(a:buf, 'lsp_did_semantic_setup', 1)
 endfunction
 
-function! s:add_highlight(server, buf, line, highlights) abort
+function! s:hash(str) abort
```

Comment:
Not sure if there is a faster way to get a hash. Since this is called many times, its execution time will dominate and we should try to make it as fast as possible.